homeassistant/components/gpsd/__init__.py

```diff
@@ -2,19 +2,41 @@
 
 from __future__ import annotations
 
+from gps3.agps3threaded import AGPS3mechanism
+
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import Platform
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
 from homeassistant.core import HomeAssistant
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up GPSD from a config entry."""
+    agps_thread = AGPS3mechanism()
+    entry.runtime_data = agps_thread
+
+    await hass.async_add_executor_job(
+        agps_thread.stream_data,
+        entry.data.get(CONF_HOST),
+        entry.data.get(CONF_PORT),
+    )
+    await hass.async_add_executor_job(agps_thread.run_thread)
```

Comment:
Let's not make 2 executor jobs at once, we can merge this into one job with a function

Reply:
Done, thanks :)